### PR TITLE
Python: Add non-alert data for lines of code and extraction errors

### PR DESCRIPTION
**Pull request by RasmusWL**

### CODE REVIEW

File: python/ql/src/Summary/LinesOfCode.ql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @id py/summary/lines-of-code
+ * @name Total lines of Python code in the database
+ * @description The total number of lines of Python code across all files, including
+ *   external libraries and auto-generated files. This is a useful metric of the size of a
+ *   database. This query counts the lines of code, excluding whitespace or comments.
+ * @kind metric
+ * @tags summary
+ */
```
Comment (tausbn) [inline python/ql/src/Summary/LinesOfCode.ql:9]:
```suggestion
/**
 * @name Total lines of Python code in the database
 * @description The total number of lines of Python code across all files, including
 *   external libraries and auto-generated files. This is a useful metric of the size of a
 *   database. This query counts the lines of code, excluding whitespace or comments.
 * @kind metric
 * @tags summary
 * @id py/summary/lines-of-code
 */
```
Oh, what a terrible diff this produced. All I did was move the first line to the end... :facepalm:

File: python/ql/src/Summary/LinesOfUserCode.ql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @id py/summary/lines-of-user-code
+ * @name Total lines of user written Python code in the database
+ * @description The total number of lines of Python code from the source code directory,
+ *   excluding auto-generated files. This query counts the lines of code, excluding
+ *   whitespace or comments. Note: If external libraries are included in the codebase
+ *   either in a checked-in virtual environment or as vendored code, that will currently
+ *   be counted as user written code.
+ * @kind metric
+ * @tags summary
```
Comment (tausbn) [inline python/ql/src/Summary/LinesOfUserCode.ql:10]:
```suggestion
 * @name Total lines of user written Python code in the database
 * @description The total number of lines of Python code from the source code directory,
 *   excluding auto-generated files. This query counts the lines of code, excluding
 *   whitespace or comments. Note: If external libraries are included in the codebase
 *   either in a checked-in virtual environment or as vendored code, that will currently
 *   be counted as user written code.
 * @kind metric
 * @tags summary
 * @id py/summary/lines-of-user-code
```
As above.

**Review by tausbn — DISMISSED** [review]
Short and sweet. Nicely done! :+1: 
I have made some suggestions, putting two wayward `@id` tags in a more familiar place. Apart from that this looks good to go.

**Comment by tausbn** [comment]
Looking at the test failures, I fear you'll have to split up the `ExtractionErrors.ql` test into two versions (for Python 2 and 3 respectively) as it seems they have different names for `utf-8`. :man_facepalming:

**Comment by tausbn** [comment]
Looking at the test failures, I fear you'll have to split up the `ExtractionErrors.ql` test into two versions (for Python 2 and 3 respectively) as it seems they have different names for `utf-8`. :man_facepalming:

File: python/ql/src/Summary/LinesOfCode.ql (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/Summary/LinesOfCode.ql:9]: I did consider doing this as well, since it looked strange after I had copied the files :stuck_out_tongue:

**Review by tausbn — APPROVED** [review]

**Review by tausbn — APPROVED** [review]

**Comment by RasmusWL** [comment]
These queries are still a bit too bleeding edge, and currently fails with `WARNING: Unrecognized value 'diagnostic' for the 'kind' property` (since we treat any warnings as errors)

**Comment by RasmusWL** [comment]
Internal tools have been updated now to allow this `@kind` :+1: :shipit:

**Pull request merged**